src/include/storage/index/index_builder.h

```diff
@@ -13,6 +14,8 @@ class IndexMetadata;
 class IndexBuilder {
```

Comment:
`IndexBuilder` is a [Builder](https://en.wikipedia.org/wiki/Builder_pattern) to generate an index from a given schema. This logic is relatively complicated (different key types, underlying data structures, etc.), hence this abstraction. It should not be used to perform index population. We need to maintain a clean separation of concerns.